go-controller/pkg/ovn/pods.go

```diff
@@ -300,46 +262,68 @@ func (oc *Controller) addLogicalPort(pod *kapi.Pod) error {
 			"--", "--if-exists", "clear", "logical_switch_port", portName, "dynamic_addresses",
 		)
 	} else {
-		addresses = "dynamic"
-		networks, err := util.GetPodNetSelAnnotation(pod, util.DefNetworkAnnotation)
-		if err != nil || (networks != nil && len(networks) != 1) {
+		podMac, podIfAddrs, err = oc.getPortAddresses(logicalSwitch, portName)
```

Comment:
this is the corner case that @squeed had pointed out where the annotations are not set on the pod, but the ovndb entries are present and then ovnkube process crashes. So we get addresses from the lsp in the db first otherwise call `assignPodAddresses`